src/main/java/org/thethingsnetwork/java/app/lib/Message.java

```diff
@@ -1,39 +1,45 @@
 /*
- * Shareif.com CONFIDENTIAL
- * ________________________
+ * The MIT License
  *
- * Copyright 2016 Shareif.com SPRL
- * All Rights Reserved.
+ * Copyright (c) 2016 The Things Network
  *
- * NOTICE:  All information contained herein is, and remains
- * the property of Shareif.com SPRL and its suppliers,
- * if any.  The intellectual and technical concepts contained
- * herein are proprietary to Shareif.com SPRL
- * and its suppliers and may be covered by Belgian and Foreign Patents,
- * patents in process, and are protected by trade secret or copyright law.
- * Dissemination of this information or reproduction of this material
- * is strictly forbidden unless prior written permission is obtained
- * from Shareif.com SPRL.
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
  */
 package org.thethingsnetwork.java.app.lib;
 
 import java.util.Base64;
 import org.json.JSONException;
 import org.json.JSONObject;
-import static org.json.JSONObject.quote;
 
 /**
- * This is a wrapper class for JsonObject to provide support for base64-encoded data
- * @author Romain Cambier <romain@shareif.com>
+ * This is a wrapper class for JSONObject to provide support for base64 encoded payload
+ *
+ * @author Romain Cambier
  */
 public class Message extends JSONObject {
 
-    public byte[] getBinary(String _key) {
-        Object object = get(_key);
+    public byte[] getRawPayload() {
```

Comment:
I finally see how JsonObject works very different from other implementations for other languages I've seen and that you have to use a type specific getter and also go [all weirdo](http://stackoverflow.com/a/14751649) to get deep fields. I get why you did `getBinary()` now and that it is probably more Java-like then what you changed it to now. So let's change it back. Sorry 🙃